core/pipeline/label.go

```diff
@@ -139,12 +140,29 @@ func LoadGitLabels(workdir string) ([]Label, error) {
 		return nil, err
 	}
 
+	// Check if the commit is a merge commit (GitHub context)
+	if commit.NumParents() > 1 {
```

Comment:
So wouldn't this also trigger on main, when doing merge commits (instead of rebase/squash)?

In that case, the message *should* be the merge commit, right? (since it was the last commit on that branch)

Can we use something else to identify this magic commit? I can't find any docs on this, but I wonder if the commit message has more details, or whether the commit author is an automated github thing?

Reply:
> So wouldn't this also trigger on main, when doing merge commits (instead of rebase/squash)?

I don't think so since here https://github.com/dagger/dagger/pull/6253/files#diff-8b206da82ce6c3970baf564e295175e55119318f3a46ca53d8b90dde0b266b34R144-R145 we're specifically checking if this is a Github PR or Gitlab MR.